packages/react-core/src/demos/DataList/examples/DataListPagination.tsx

```diff
@@ -0,0 +1,315 @@
+  OverflowMenuItem,
+  PageSection,
+  PageSectionVariants,
+  Pagination,
+  Text,
+  TextContent,
+  TextVariants,
+  Toolbar,
+  ToolbarItem,
+  ToolbarContent
+} from '@patternfly/react-core';
+import EllipsisVIcon from '@patternfly/react-icons/dist/esm/icons/ellipsis-v-icon';
+import { DashboardWrapper } from '@patternfly/react-core/dist/js/demos/DashboardWrapper';
+
+import CodeBranchIcon from '@patternfly/react-icons/dist/esm/icons/code-branch-icon';
+import CodeIcon from '@patternfly/react-icons/dist/esm/icons/code-icon';
+import CubeIcon from '@patternfly/react-icons/dist/esm/icons/cube-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+import TimesCircleIcon from '@patternfly/react-icons/dist/esm/icons/times-circle-icon';
+
+export const DataListPagination: React.FunctionComponent = () => {
+  const renderPagination = () => <Pagination itemCount={5} page={1} variant="top" isCompact />;
+
+  const toolbarItems = (
+    <React.Fragment>
+      <ToolbarItem variant="bulk-select">
+        <MenuToggle
+          aria-label="Select cards"
+          splitButtonOptions={{
+            items: [
+              <MenuToggleCheckbox
+                id="split-dropdown-checkbox"
+                key="split-dropdown-checkbox"
+                aria-label={'Select all cards'}
+              />
+            ]
+          }}
+        ></MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem>
+        <MenuToggle>Filter by creator name</MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem variant="overflow-menu">
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuItem>
+            <Button variant="primary">Create instance</Button>
+          </OverflowMenuItem>
+          <OverflowMenuControl hasAdditionalOptions>
+            <MenuToggle aria-label="Toolbar kebab overflow menu" variant="plain">
+              <EllipsisVIcon />
+            </MenuToggle>
+          </OverflowMenuControl>
+        </OverflowMenu>
+      </ToolbarItem>
+      <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+        {renderPagination()}
+      </ToolbarItem>
+    </React.Fragment>
+  );
+
+  return (
+    <React.Fragment>
+      <DashboardWrapper mainContainerId="main-content-datalist-view-default-nav" breadcrumb={null}>
+        <PageSection variant={PageSectionVariants.light}>
+          <TextContent>
+            <Text component="h1">Projects</Text>
+            <Text component="p">This is a demo that showcases PatternFly Data List</Text>
+          </TextContent>
+        </PageSection>
+        <PageSection isFilled>
+          <Toolbar id="toolbar-group-types">
+            <ToolbarContent>{toolbarItems}</ToolbarContent>
+          </Toolbar>
+          <DataList aria-label="Demo data list">
+            <DataListItem aria-labelledby="Demo-item1">
+              <DataListItemRow>
```

Comment:
i think if i were to come into this demo, i'd be expecting more "show me how to use pagination", and there is a huge block of code just for displaying the datalist.

i think we could have more simple items that are create with a loop.